### PR TITLE
fix: update channel info when members change

### PR DESCRIPTION
**Pull request by guidota**
# What? 
Bump to `matrix-client` to version `1.18.0 `

Listen for channel members updates and notify the updates to explorer

Send joined members count on channel leave action

Send online members count on get joined channels action

### CODE REVIEW

File: packages/shared/friends/sagas.ts [hunk 1]
```diff
@@ -389,18 +389,18 @@ function* configureMatrixClient(action: SetMatrixClient) {
     handleIncomingFriendshipUpdateStatus(FriendshipAction.REJECTED, socialId)
   )
 
+  client.onChannelMembers((conversation) => {
+    updateChannelInfo(conversation, client)
```
Comment (Julieta11) [inline packages/shared/friends/sagas.ts:393]: Shouldn't we also send the update about the members list with the `UpdateChannelMembers` message? I think that's the message that will solve the blank member list and toggle issue.

**Review by Julieta11 — APPROVED** [review]
🙌🏼 🏃🏼‍♀️! I just left a quick question comment

File: packages/shared/friends/sagas.ts [hunk 2]
```diff
@@ -389,18 +389,26 @@ function* configureMatrixClient(action: SetMatrixClient) {
     handleIncomingFriendshipUpdateStatus(FriendshipAction.REJECTED, socialId)
   )
 
+  client.onChannelMembers((conversation, members) => {
+    updateChannelInfo(conversation, client)
```
Comment (Julieta11) [inline packages/shared/friends/sagas.ts:393]: I think before we do anything we can make use of this validation here: 
```
if (!conversation.name || conversation.name?.startsWith('Empty room')) {
          break
}
```

File: packages/shared/friends/sagas.ts [hunk 3]
```diff
@@ -389,18 +389,28 @@ function* configureMatrixClient(action: SetMatrixClient) {
...
 
+  client.onChannelMembers((conversation, members) => {
+    if (conversation.name && !conversation.name.startsWith('Empty room')) {
+      updateChannelInfo(conversation, client)
+    }
+
+    const userStatuses = client.getUserStatuses(...members.map((member) => member.userId))
+    const membersPayload = members
+      .filter((member) => userStatuses.get(member.userId)?.presence === PresenceType.ONLINE)
+      .map((member) => ({ ...member, isOnline: true }))
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:400]: I'm wondering if there's a way we can reuse the logic from the `getOnlineMembers` or some intermediate function.

**Review by menduz — DISMISSED** [review]

File: packages/shared/friends/sagas.ts [hunk 4]
```diff
@@ -389,18 +389,39 @@ function* configureMatrixClient(action: SetMatrixClient) {
...
+
+    const channelMembers: ChannelMember[] = members
+      .filter((member) => onlineMemberIds.includes(member.userId))
+      .map((member) => ({
+        userId: getUserIdFromMatrix(member.userId),
+        name: member.name,
+        isOnline: true
+      }))
+
+    const update: UpdateChannelMembersPayload = { channelId: conversation.id, members: channelMembers }
```
Comment (menduz) [inline packages/shared/friends/sagas.ts:411]: for future features, I'd recommend to put this logic inside dcl-matrix-client

**Review by menduz — APPROVED** [review]

**Pull request merged**